### ISSUE TITLE
Add a Differentially-Private AdaBoost model

### ISSUE THREAD

**Issue by UPstartDeveloper** [IBM/differential-privacy-library#83]
**Is your feature request related to a problem? Please describe.**
I've been trying to understand the paper, "[Efficient, Noise-Tolerant, and Private Learning via Boosting](https://arxiv.org/pdf/2002.01100.pdf)" and it is difficult as the work is purely theoretical.

**Describe the solution you'd like**
I want to have an actual implementation of the boosting framework presented in this ^paper, for the purpose of furthering my own understanding. 

**Describe alternatives you've considered**
So far it looks like [TensorFlow Decision Forests](https://www.tensorflow.org/decision_forests/api_docs/python/tfdf/builder/GradientBoostedTreeBuilder) has a module similar to what I'm looking for - but it looks like overkill to use it, subclass it, all to use a new kind of optimization algorithm (based on the paper I'm reading).

**Additional context**
- I'm interested to understand this library better, so I've actually started trying to implement this on my [fork](https://github.com/UPstartDeveloper/differential-privacy-library). So far it includes (what I think) is the code for the [model](https://github.com/UPstartDeveloper/differential-privacy-library/blob/zain-dp-adaboost/diffprivlib/models/adaboost.py)) and a demo [notebook](https://github.com/UPstartDeveloper/differential-privacy-library/blob/zain-dp-adaboost/notebooks/plot_adaboost_twoclass.ipynb).
- The main blockers I think I have right now are the following:
1) **reproducibility** - when trying to run my current implementation (i.e., in `adaboost.py`) in the notebook (see cell `12` of `plot_adaboost_twoclass.ipynb`), it always seems to "bounce around" in training accuracy. E.g. `43%` for the first run, then `47%`, and eventually `0.00%` within ~5 runs. I'm trying to understand if this is an expected result, and if not what can be done to fix it?
2) **testing** - how should I go about writing tests for the `adaboost.py`?

Any other feedback/questions are appreciated!
